From 0bcb3df2de57ad65d13820ba31ac68fbe1fa4653 Mon Sep 17 00:00:00 2001
From: Peng Liang <liangpeng10@huawei.com>
Date: Fri, 19 Feb 2021 15:14:33 +0800
Subject: [PATCH] cpuidle: haltpoll: Only check boot_option_idle_override in
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 0bcb3df2de57ad65d13820ba31ac68fbe1fa4653

 x86

hulk inclusion
category: feature
bugzilla: 47727
CVE: NA

------------------------------

boot_option_idle_override is defined only in x86/ia64.  Since haltpoll
supports x86 and arm64, let's check boot_option_idle_override only in
x86.

Signed-off-by: Peng Liang <liangpeng10@huawei.com>
Reviewed-by: Hanjun Guo <guohanjun@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 drivers/cpuidle/cpuidle-haltpoll.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/cpuidle/cpuidle-haltpoll.c b/drivers/cpuidle/cpuidle-haltpoll.c
index f2fc7204de54..f47612852582 100644
--- a/drivers/cpuidle/cpuidle-haltpoll.c
+++ b/drivers/cpuidle/cpuidle-haltpoll.c
@@ -104,9 +104,11 @@ static int __init haltpoll_init(void)
 	int ret;
 	struct cpuidle_driver *drv = &haltpoll_driver;
 
+#ifdef CONFIG_X86
 	/* Do not load haltpoll if idle= is passed */
 	if (boot_option_idle_override != IDLE_NO_OVERRIDE)
 		return -ENODEV;
+#endif
 
 	cpuidle_poll_state_init(drv);
 
-- 
2.26.2

